enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/HighAvailabilityMemberState.java

```diff
@@ -59,11 +59,9 @@ public HighAvailabilityMemberState masterIsElected( HighAvailabilityMemberContex
                 public HighAvailabilityMemberState masterIsAvailable( HighAvailabilityMemberContext context,
                                                                       InstanceId masterId, URI masterHaURI )
                 {
-//                    assert context.getAvailableMaster() == null;
                     if ( masterId.equals( context.getMyId() ) )
                     {
-                        throw new RuntimeException( "Received a MasterIsAvailable event for my InstanceId while in" +
-                                " PENDING state" );
+                        return PENDING_ELECTION;
```

Comment:
Yep that is part of initial question we send them to pending as initial state and then forcing elections in case if we where already before here with same state.
That will save us from handling one more state. If i'm not missing something :)